x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -69,6 +69,12 @@ public class AnalysisConfig implements ToXContentObject, Writeable {
     public static final String ML_CATEGORY_FIELD = "mlcategory";
     public static final Set<String> AUTO_CREATED_FIELDS = new HashSet<>(Collections.singletonList(ML_CATEGORY_FIELD));
 
+    // Since the C++ backend truncates the categorization field at length 1000, adding an ellipsis on truncation, it makes no sense to
```

Comment:
Please also mention it's in `model::CCategoryExamplesCollector::MAX_EXAMPLE_LENGTH` in case someone needs to tie up the two sides in the future.